### ISSUE TITLE
Initializing new applications

### ISSUE THREAD

**Issue by radu-matei** [spinframework/spin#16]
Currently, Spin has no concept of "initializing" a new application without using a template (re #15).

Is there a need for a `spin init` command? What would it do?

**Comment by mikkelhegn** [comment]
I can only see two reasons:
1. Do not need source, but will just reference a pre-built .wasm module for components.
2. A very opinionated directory layout is preferred.

For 1. I believe the feature to re-use components with a URL reference is the best option to promote. We could have a template supporting this scenario, where the template would require the URL and Digest as part of the template creation flow.

For 2. The only thing we coudl do was to create a `spin.toml` file with no components (a few lines). This could also be a template, e.g., `empty app`.

**Comment by itowlson** [comment]
FWIW there is a `http-empty` template, which is a `spin.toml` with no components, but it does set the application `trigger` to HTTP.

**Comment by michelleN** [comment]
@radu-matei What is the use case you're targeting for initializing a new application? What kind of workflow are you envisioning?

**Comment by itowlson** [comment]
Additional context: https://github.com/fermyon/spin/issues/1525#issuecomment-1756104398

**Comment by michelleN** [comment]
Considering we have an `--init` flag on spin new now, do we still want to have a new top level command for 2.0? Should we try out `spin new --init` for a while and come back to this post 2.0?

**Comment by itowlson** [comment]
At this stage I would be strongly in favour of not attempting this for 2.0.  With Manifest 2, it becomes a lot more practical than it was with Manifest 1 though, so getting rid of what is now a slightly artificial `http-empty` template seems like a good plan!

**Comment by calebschoepp** [comment]
Is this still something we want @radu-matei @michelleN? I happened across this issue and it feels like possibly this use case is no longer valid and we could close this issue.

**Comment by radu-matei** [comment]
Yeah, I think it's safe to close this; I haven't seen more people ask for changes to the new experience; we can reopen if needed.
